File: POO/poo11048216/Aulas/aula15/Conta/ContaInvestimento.cpp
#ifndef CONTAINVESTIMENTO_H
#include "ContaInvestimento.h"
#define CONTAINVESTIMENTO_H
#endif

ContaInvestimento::ContaInvestimento()
{   }

ContaInvestimento::ContaInvestimento(std::string cpf, std::string nome, float txR, float txA):Conta(cpf, nome)
{
    this->setTaxaRendimento(txR);
    this->setTaxaAdministracao(txA);
}

void ContaInvestimento::setTaxaRendimento(float txR)
{
    this->taxaRendimento = txR;
}

void ContaInvestimento::setTaxaAdministracao(float txA)
{
    this->taxaAdministracao = txA;
}

float ContaInvestimento::getTaxaRendimento()
{
    return this->taxaRendimento;
}

float ContaInvestimento::getTaxaAdministracao()
{
    return this->taxaAdministracao;
}

void ContaInvestimento::correcao()
{
    if(!this->isBloqueada())
    {
        this->setSaldo(this->getSaldo() * (1.0 + (this->getTaxaRendimento()/100.0)));
    }
}

void ContaInvestimento::extrato()
{
    Conta::extrato();
    std::cout << "TAXA DE RENDIMENTO ....: " << this->getTaxaRendimento() << "\%a.m" << std::endl;
    std::cout << "TAXA DE ADMINISTRACAO .: " << this->getTaxaAdministracao() << "\%a.m" << std::endl;
    std::cout << "===================================== " << std::endl;
}

File: POO/poo11048216/Aulas/aula16.1/JConta/ContaInvestimento.java
class ContaInvestimento extends Conta
{
    private float taxaRendimento;
    private float taxaAdministracao;
    
    public ContaInvestimento()
    {   }
    
    public ContaInvestimento(String cpf, String nome, float txR, float txA)
    {
        super(cpf, nome);
        this.setTaxaRendimento(txR);
        this.setTaxaAdministracao(txA);
    }
    
    public void setTaxaRendimento(float tx)
    {
        this.taxaRendimento = tx;
    }
    
    public void setTaxaAdministracao(float tx)
    {
        this.taxaAdministracao = tx;
    }
    
    public float getTaxaRendimento()
    {
        return this.taxaRendimento;
    }
    
    public float getTaxaAdministracao()
    {
        return this.taxaAdministracao;
    }
    
    public void correcao()
    {
        this.deposito(this.getSaldo()*(this.getTaxaRendimento()/100.0f));
    }
    
    public void extrato()
    {
        super.extrato();
        System.out.println("TAXA DE RENDIMENTO ....: " + this.getTaxaRendimento() + "/% a.m.");
        System.out.println("TAXA DE ADMINISTRACAO .: " + this.getTaxaAdministracao() + "/% a.m.");
        System.out.println("===================================");
    }
}